### PR TITLE
Save scenario in game data

### PR DESCRIPTION
**Pull request by daeglin**

### CODE REVIEW

**Comment by jnwltr** [comment]
I'll take this one, check your changes, and follow up with my validation-related tasks.

File: backend/src/game/routes.ts [hunk 1]
```diff
@@ -77,7 +77,9 @@ function validate(data: any): boolean {
   if (!data.simulation || !data.simulation.length) return false;
 
   try {
-    if (!validateGame(data)) return false;
+    const game = validateGame(data);
+    if (!game) return false;
+    if (game.isGameLost() || !game.isFinished()) return false;
```
Comment (jnwltr) [inline backend/src/game/routes.ts:82]: I think this one should be inside `validateGame`, what do you think?

File: frontend/src/app/services/validate.ts [hunk 2]
```diff
@@ -38,5 +36,7 @@ export function validateGame(data: GameData, breakImmediately = true): Game | un
 
   if (!res) console.error(`Validation failed, new data: ${JSON.stringify(game.simulation.modelStates, null, 2)}`);
 
+  // Set the real scenario for the future gameplay mitigations
+  game.scenario = new Scenario(data.scenario.dates, data.scenario.gameplayMitigationHistory);
```
Comment (jnwltr) [inline frontend/src/app/services/validate.ts:40]: I kind of dislike this duality. And also the duality of gameplay and ramp-up mitigations. Could not we join these pairs somehow? To be discussed offline...

**Review by jnwltr — COMMENTED** [review]
Pls, let's discuss my comments/outlooks for further steps.

File: backend/src/game/routes.ts (hunk 1 above)
Comment (daeglin) [reply backend/src/game/routes.ts:82]: We use `valiadateGame` in outro (when loading game by id). I think this is fine. validateGame checks the consistency of data. Server does some extra checks whether the game is can be displayed in the leaderboard (uses legit scenario etc)

**Pull request closed without merging**